sdk_extensions/aws_v1_support/src/main/java/io/opentelemetry/sdk/extensions/trace/aws/resource/AwsResource.java

```diff
@@ -31,25 +30,26 @@
    * environment, e.g., metadata for the instance if the app is running on EC2.
    */
   public static Resource create() {
-    return create(new Ec2Resource());
+    return create(new Ec2Resource(), new EcsResource(), new BeanstalkResource());
   }
 
   @VisibleForTesting
   static Resource create(AwsResource... populators) {
-    Map<String, AttributeValue> attrMap = new LinkedHashMap<>();
-
+    final Attributes.Builder attrBuilder = Attributes.newBuilder();
     for (AwsResource populator : populators) {
-      attrMap.putAll(populator.createAttributes());
-    }
-
-    Attributes.Builder attrBuilder = Attributes.newBuilder();
-    for (Map.Entry<String, AttributeValue> attr : attrMap.entrySet()) {
-      attrBuilder.setAttribute(attr.getKey(), attr.getValue());
+      Attributes attrs = populator.createAttributes();
+      attrs.forEach(
```

Comment:
This makes me think that a "merge" function might be useful on the Attributes, like we have on the Resource itself.

Reply:
A ready-to-use `merge` would be better. One more thing I found in unit test is if builder set the duplicated key, the final Attributes has the fist one but not the last one. It seems wired that usually the last one can override the first. Please see my AwsResourceTest.java

Reply:
Yes, we currently keep the first, if there are duplicates. If you think this is not correct, please file an issue!